### PR TITLE
[IMP] html_editor: apply color and font-size to list item marker

### PR DESCRIPTION
**Pull request by deso-odoo**
Description of the issue this PR addresses:

Current behavior before PR:

It was not possible to apply color and font-size to list item markers.

Desired behavior after PR is merged:

This commit ensures that if a list item is selected completely, applying color and font-size to it will apply them to the list item itself making sure the list item marker are also gets styled.

Note:
I. The color and font-size is applied to marker only when the entire list item is selected, not if list item is only partially selected.
II. The font-size property will not apply to the marker also if the list item contains block elements (eg. H1, H2, H3, ...etc), as applying font-size to list item sets `list-style-position: inside` which moves block elements to next line.

task-3281223

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/html_editor/static/src/main/list/list_plugin.js [hunk 1]
```diff
@@ -835,4 +851,79 @@ export class ListPlugin extends Plugin {
             }
         }
     }
+
+    applyListFormat(formatName, { formatProps } = {}) {
```
Comment (dmo-odoo) [inline addons/html_editor/static/src/main/list/list_plugin.js:855]: Unless I'm mistaken, this function does not apply a list format, it applies a format to a list, correct? Therefore I would name it `applyFormatToList` instead.

**Review by dmo-odoo — COMMENTED** [review]
@deso-odoo Impressive work! I only wrote a tiny comment about one of the function names which I'm not sure is adequate.

File: addons/html_editor/static/src/main/list/list_plugin.js (hunk 1 above)
Comment (deso-odoo) [reply addons/html_editor/static/src/main/list/list_plugin.js:855]: You are correct we apply apply format to a list, particularly list item. How about `applyFormatToListItem` or should i just keep it `applyFormatToList` also the same is true for `applyListColor` shall I change this too?

Comment (dmo-odoo) [reply addons/html_editor/static/src/main/list/list_plugin.js:855]: Yes, `applyFormatToListItem` and `applyColorToListItem` sound great, thanks!

Comment (deso-odoo) [reply addons/html_editor/static/src/main/list/list_plugin.js]: Done!

**Comment by deso-odoo** [comment]
> @deso-odoo Impressive work!

:see_no_evil: I was expecting tons of comments. ^^'

**Comment by dmo-odoo** [comment]
@deso-odoo Maybe I can find stuff to nitpick about if you really want? 😛 Otherwise I'll r+ it

**Comment by deso-odoo** [comment]
@dmo-odoo :eyes: I'll leave that to you :grin:

**Comment by dmo-odoo** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**